_wasm_crypto/algorithms.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * All cryptographic hash/digest algorithms supported by std/_wasm_crypto.
+ *
+ * For algorithms that are supported by WebCrypto, the name here must match the
+ * one used by WebCrypto. Otherwise we should prefer the formatting used in the
+ * official specification. All names are uppercase to faciliate case-insensitive
+ * comparisons required by the WebCrypto spec.
+ */
+export const digestAlgorithms = [
+  "BLAKE2B-256",
+  "BLAKE2B-384",
+  "BLAKE2B",
+  "BLAKE2S",
+  "BLAKE3",
+  "KECCAK-224",
+  "KECCAK-256",
+  "KECCAK-384",
+  "KECCAK-512",
+  "SHA-384",
+  "SHA3-224",
+  "SHA3-256",
+  "SHA3-384",
+  "SHA3-512",
+  "SHAKE128",
+  "SHAKE256",
+  // insecure (length-extendable):
+  "RIPEMD-160",
+  "SHA-224",
+  "SHA-256",
+  "SHA-512",
+  // insecure (collidable and length-extendable):
+  "MD5",
+  "SHA-1",
+] as const;
+
+/** A cryptographic hash/digest algorithm supported by std/_wasm_crypto. */
+export type DigestAlgorithm = typeof digestAlgorithms[number];
+
+/**
+ * Other platforms may identify hash algorithms by slightly different names,
+ * so we use this map of some common variations to the names we support.
+ */
+export const digestAliases: Record<string, DigestAlgorithm> = {
```

Comment:
I think there needs to be a good reason why we would alias them. I think we should provide a canonical name.  Allowing aliases leads to accidents and we don't really want accidents with security related software. If the canonical algorithm name isn't available, we should just throw, maybe with a list of supported alogrithms.

Reply:
I've moved this into `node/crypto` and pared it down to differently-formatted hash identifiers I saw from Node's `.getHashes()`.